File: sol/solution/0001-0100/0003. Longest Substring Without Repeating Characters/Solution.rs
use std::collections::HashMap;
impl Solution {
    pub fn length_of_longest_substring(s: String) -> i32 {
        let mut last = HashMap::new();
        let (mut res, mut start) = (0, 0);
        let cs: Vec<char> = s.chars().collect();
        for (i, c) in cs.iter().enumerate() {
            if let Some(&idx) = last.get(c) {
                if idx >= start {
                    start = idx + 1;
                }
            }
            res = res.max(i - start + 1);
            last.insert(*c, i);
        }
        res as i32
    }
}

File: sol/solution/0001-0100/0003. Longest Substring Without Repeating Characters/Solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        last = {}
        res = start = 0
        for i, c in enumerate(s):
            if c in last and last[c] >= start:
                start = last[c] + 1
            res = max(res, i - start + 1)
            last[c] = i
        return res